app/models/user.rb

```diff
@@ -3,6 +3,16 @@ class User < ApplicationRecord
   USERS_PARAMS = %i(name email password password_confirmation).freeze
 
   has_many :microposts, dependent: :destroy
+  has_many :active_relationships, class_name: Relationship.name,
+    foreign_key: :follower_id,
+    dependent: :destroy,
+    inverse_of: :follower
+  has_many :passive_relationships, class_name: Relationship.name,
+    foreign_key: :followed_id,
+    dependent: :destroy,
+    inverse_of: :followed
+  has_many :following, through: :active_relationships, source: :followed
```

Comment:
has_many :through để set up quan hệ many-to-many giữa 2 đối tượng trong Rails thông qua 1 bên thứ ba, mặc định tìm đến khoá ngoại theo convention, VD ở đây sẽ là following_id, tuy nhiên vì ta đã set up các khoá ngoại trước đó nên phải dùng tuỳ chọn source để chỉ định các khoá ngoại này